presto-main/src/main/java/com/facebook/presto/sql/analyzer/FeaturesConfig.java

```diff
@@ -2915,4 +2916,17 @@ public FeaturesConfig setSkipHashGenerationForJoinWithTableScanInput(boolean ski
         this.skipHashGenerationForJoinWithTableScanInput = skipHashGenerationForJoinWithTableScanInput;
         return this;
     }
+
+    public boolean getGenerateDomainFilters()
+    {
+        return generateDomainFilters;
+    }
+
+    @Config("optimizer.generate-domain-filters")
+    @ConfigDescription("Infer column domains during predicate pushdown through joins and add them as filters")
```

Comment:
Also IMO "domain filters" may not be obvious to the lay user. Maybe "infer-pushdown-predicates-from-complex-join-filters" or such like?

Reply:
>Also IMO "domain filters" may not be obvious to the lay user.
I think we can introduce this nomenclature here since we already do this with other optimizer internals, e.g for OPTIMIZE_HASH_GENERATION, we have 'Compute hash codes for distribution, joins, and aggregations early in query plan'

To ease the transition, I was going to add a more verbose description in properties.rst, like we do [here](https://prestodb.io/docs/current/admin/properties.html#optimizer-optimize-hash-generation)

>The part that feels most unclear is ... during predicate pushdown through joins.... Reading it is difficult to parse initially for me IMO.
How about we go minimal with the description - 'Infer predicates from column domains during predicate pushdown' here in code and add a more verbose description in properties.rst

Reply:
> How about we go minimal with the description - 'Infer predicates from column domains during predicate pushdown'

Sounds good to me